src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -957,6 +957,19 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
     if (!created)   /// Table already exists
         return {};
 
+    /// Check type compatible for materialized dest table and select columns
+    if (create.select && create.is_materialized_view && create.to_table_id)
+    {
+        StoragePtr table = DatabaseCatalog::instance().getTable({create.database, create.table, create.uuid}, getContext());
+        const auto & output_columns = table->getInMemoryMetadataPtr()->getSampleBlock();
+        Block input_columns=InterpreterSelectWithUnionQuery(
+            create.select->clone(), getContext(),SelectQueryOptions().analyze()).getSampleBlock();
```

Comment:
```suggestion
        Block input_columns = InterpreterSelectQuery(
            create.select->clone(), getContext(), SelectQueryOptions().analyze()).getSampleBlock();
```

- coding style issues
- and seems that `InterpreterSelectQuery` is enough?

P.S. untested.

Reply:
I tried to use `InterpreterSelectQuery `, but when run test `01182_materialized_view_different_structure` it failed.